workerpool/pool.go

```diff
@@ -21,34 +23,37 @@ import (
 // (configured by connection.Job struct). Each job runs one or more workers. Each worker processes
 // one shard/partition from the connection. Worker pool run jobs based on kv.Event channel.
 type WorkerPool struct {
-	maxWorkers  uint
-	numWorkers  uint
-	locksPrefix string
-	session     *concurrency.Session
-	events      <-chan *kv.Event
-	jobs        map[connection.JobID]*job // map of job handlers assigned to each connection.ID
-	jobsMutex   sync.RWMutex
-	log         *zap.SugaredLogger
+	maxWorkers   uint
+	numWorkers   uint
+	locksPrefix  string
+	checkpointKV *kv.Store
```

Comment:
that creates an unnecessary dependency on `kv.Store` struct. I think that the better approach would be creating an `Checkpointer` interface with two methods and using `kv.Store` as an implementation.

Reply:
ok, i moved this only because of the `deleteWorkers` attempt, will revert